### PR TITLE
[baseimage]: upgrade base image to debian buster

### PR DESCRIPTION
**Pull request by lguohan**
**- What I did**
bring up the base image to debian buster 4.19 kernel.

This pr depends on the linux-kernel pr to be merged first.

**- How I did it**

**- How to verify it**
build vs image and tested.

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: build_debian.sh [hunk 1]
```diff
@@ -481,8 +482,9 @@ if [ "${enable_organization_extensions}" = "y" ]; then
 fi
 
 ## Setup ebtable rules (rule file is in binary format)
-sudo sed -i 's/EBTABLES_LOAD_ON_START="no"/EBTABLES_LOAD_ON_START="yes"/g' ${FILESYSTEM_ROOT}/etc/default/ebtables
-sudo cp files/image_config/ebtables/ebtables.filter ${FILESYSTEM_ROOT}/etc
+# TO fix in buster
+# sudo sed -i 's/EBTABLES_LOAD_ON_START="no"/EBTABLES_LOAD_ON_START="yes"/g' ${FILESYSTEM_ROOT}/etc/default/ebtables
+# sudo cp files/image_config/ebtables/ebtables.filter ${FILESYSTEM_ROOT}/etc
```
Comment (lguohan) [inline build_debian.sh:487]: @lguohan  to address

File: files/image_config/misc/docker-wait-any [hunk 2]
```diff
@@ -11,13 +11,12 @@
     stopped running. Here, we spawn multiple threads and wait on one
     container per thread. If any of the threads exit, the entire
     application will exit.
-    NOTE: This script is written against docker-py version 1.6.0. Newer
-    versions of docker-py have a different API.
+    NOTE: This script is written against docker 4.1.0.
```
Comment (lguohan) [inline files/image_config/misc/docker-wait-any:14]: @jleveque to check.

File: .gitmodules [hunk 3]
```diff
@@ -3,7 +3,7 @@
 	url = https://github.com/Azure/sonic-swss-common
 [submodule "sonic-linux-kernel"]
 	path = src/sonic-linux-kernel
-	url = https://github.com/Azure/sonic-linux-kernel
+	url = https://github.com/lguohan/sonic-linux-kernel
```
Comment (qiluo-msft) [inline .gitmodules:6]: >lguohan [](start = 26, length = 7)

Use Azure repo?

Comment (lguohan) [reply .gitmodules:6]: kernel has been merged yet. Will merge it before commit this pr.

File: files/build_templates/sonic_debian_extension.j2 [hunk 4]
```diff
@@ -401,7 +401,7 @@ sudo LANG=C chroot $FILESYSTEM_ROOT docker $SONIC_NATIVE_DOCKERD_FOR_DOCKERFS ta
 sudo LANG=C chroot $FILESYSTEM_ROOT docker $SONIC_NATIVE_DOCKERD_FOR_DOCKERFS tag {{imagename}}:latest {{imagebasename}}:latest
 {% endif %}
 {% endfor %}
-sudo umount $FILESYSTEM_ROOT/target
+sudo LANG=C chroot $FILESYSTEM_ROOT umount target
```
Comment (qiluo-msft) [inline files/build_templates/sonic_debian_extension.j2:404]: >chroot [](start = 12, length = 6)

It is mounted without chroot
```
sudo mount --bind target $FILESYSTEM_ROOT/target
```
Why umount with chroot?

File: src/initramfs-tools/loopback-file-system-support.patch [hunk 5]
```diff
@@ -13,11 +13,29 @@ And added NFS support! Full patch below...
 live well,
   vagrant
 
+---
+ init              | 12 ++++++++++++
```
Comment (lguohan) [inline src/initramfs-tools/loopback-file-system-support.patch:17]: this is to adapt the loopback file system support patch to v0.133. the main changes are due to the introduction of shellcheck.

File: files/image_config/misc/docker-wait-any (hunk 2 above)
Comment (qiluo-msft) [inline files/image_config/misc/docker-wait-any:14]: >docker [](start = 41, length = 6)

docker-py?

File: files/image_config/ntp/ntp-systemd-wrapper [hunk 6]
```diff
@@ -0,0 +1,38 @@
+#!/bin/sh
```
Comment (qiluo-msft) [inline files/image_config/ntp/ntp-systemd-wrapper:1]: Is it related to buster? If not, separate it into another PR?

File: files/initramfs-tools/arista-convertfs.j2 [hunk 7]
```diff
@@ -224,7 +224,7 @@ cmd="wait_for_root_dev"
 run_cmd "$cmd" "$err_msg"
 
 err_msg="Error: formatting to ext4 failed"
-cmd="mke2fs -t ext4 -F -O '^huge_file,^metadata_csum' $root_dev"
+cmd="/usr/local/sbin/mke2fs -t ext4 -F -O '^huge_file,^metadata_csum' $root_dev"
```
Comment (qiluo-msft) [inline files/initramfs-tools/arista-convertfs.j2:227]: >/usr/local/sbin/ [](start = 5, length = 16)

Is it related to buster? If not, separate it into another PR?

File: files/image_config/interfaces/interfaces.j2 [hunk 8]
```diff
@@ -76,20 +76,13 @@ iface eth0 {{ 'inet' if prefix | ipv4 else 'inet6' }} static
     up ip {{ '-4' if prefix | ipv4 else '-6' }} route add default via {{ MGMT_INTERFACE[(name, prefix)]['gwaddr'] }} dev eth0 table {{ vrf_table }} metric 201
     up ip {{ '-4' if prefix | ipv4 else '-6' }} route add {{ prefix | network }}/{{ prefix | prefixlen }} dev eth0 table {{ vrf_table }}
     up ip {{ '-4' if prefix | ipv4 else '-6' }} rule add from {{ prefix | ip }}/{{ '32' if prefix | ipv4 else '128' }} table {{ vrf_table }}
-{% if (MGMT_VRF_CONFIG) and (MGMT_VRF_CONFIG['vrf_global']['mgmtVrfEnabled'] == "true") %}
```
Comment (qiluo-msft) [inline files/image_config/interfaces/interfaces.j2:79]: >MGMT_VRF_CONFIG [](start = 7, length = 15)

Is it related to buster? If not, separate it into another PR?

File: files/image_config/ntp/ntp-systemd-wrapper (hunk 6 above)
Comment (lguohan) [reply files/image_config/ntp/ntp-systemd-wrapper:1]: yes, it is. it is due to buster image change from init script to systemd wrapper.

File: files/initramfs-tools/arista-convertfs.j2 (hunk 7 above)
Comment (lguohan) [reply files/initramfs-tools/arista-convertfs.j2:227]: check explanation in the commit message. https://github.com/Azure/sonic-buildimage/pull/4405/commits/8092e9826b167cc84378564659e5e5770b04d87a

/usr/sbin/mke2fs is now from busybox which does not support
the operation we need

File: files/image_config/misc/docker-wait-any (hunk 2 above)
Comment (lguohan) [reply files/image_config/misc/docker-wait-any:14]: no. the project is named as docker now. check out this one. https://pypi.org/project/docker/

File: files/build_templates/sonic_debian_extension.j2 (hunk 4 above)
Comment (lguohan) [reply files/build_templates/sonic_debian_extension.j2:404]: check the failed here. https://sonic-jenkins.westus2.cloudapp.azure.com/job/vs/job/buildimage-vs-image-buster/43/console

```
06:58:13.578  + sudo umount ./fsroot/target
06:58:13.578  umount: ./fsroot/target: umount failed: No such file or directory.
```

I am not sure why it failed in the first time.

Comment (lguohan) [reply files/build_templates/sonic_debian_extension.j2:404]: let me add some more debug information to understand why

File: files/image_config/interfaces/interfaces.j2 (hunk 8 above)
Comment (lguohan) [reply files/image_config/interfaces/interfaces.j2:79]: yes, it is related to buster. we have removed the cgset/cgcreate tools in buster image.

File: files/build_templates/sonic_debian_extension.j2 (hunk 4 above)
Comment (lguohan) [reply files/build_templates/sonic_debian_extension.j2:404]: it looks like it does not have mtab file

```
+ sudo LANG=C chroot ./fsroot docker tag docker-syncd-vs:latest docker-syncd-vs:buster.0-dirty-20200409.090820
+ sudo mount
mount: failed to read mtab: No such file or directory
```

**Comment by lguohan** [comment]
retest broadcom please

File: build_debian.sh [hunk 9]
```diff
@@ -409,14 +412,8 @@ set /files/etc/sysctl.conf/net.core.somaxconn 512
...
 
-if [[ $CONFIGURED_ARCH == amd64 ]]; then
-    # Configure mcelog to log machine checks to syslog
-    sudo sed -i 's/^#syslog = yes/syslog = yes/' $FILESYSTEM_ROOT/etc/mcelog/mcelog.conf
-fi
-
-## docker-py is needed by Ansible docker module
-sudo https_proxy=$https_proxy LANG=C chroot $FILESYSTEM_ROOT easy_install pip
-sudo https_proxy=$https_proxy LANG=C chroot $FILESYSTEM_ROOT pip install 'docker-py==1.6.0'
+## docker is needed by Ansible docker module
```
Comment (jleveque) [inline build_debian.sh:415]: Is it confirmed that Ansible's "docker" module works with the "docker" Python package (rather than "docker-py")?

If so, I suggest change wording to `## docker Python API package is needed by Ansible docker module`

File: files/image_config/misc/docker-wait-any (hunk 2 above)
Comment (jleveque) [reply files/image_config/misc/docker-wait-any:14]: According to API docs, this looks good. However, I suggest adding the rest of my note back in the comment, as follows:

```
    NOTE: This script is written against 'docker' Python package version 4.1.0.
    Newer versions of docker may have a different API.
    
````

File: Makefile [hunk 10]
```diff
@@ -1,23 +1,36 @@
 # SONiC make file
 
 NOJESSIE ?= 0
```
Comment (jleveque) [inline Makefile:3]: Can we eliminate references to Jessie at this point?

File: build_debian.sh [hunk 11]
```diff
@@ -184,11 +184,6 @@ if [[ $CONFIGURED_ARCH == armhf || $CONFIGURED_ARCH == arm64 ]]; then
     cat files/initramfs-tools/modules.arm | sudo tee -a $FILESYSTEM_ROOT/etc/initramfs-tools/modules > /dev/null
 fi
 
-if [[ $CONFIGURED_ARCH == amd64 ]]; then
-    ## Install latest intel ixgbe driver
-    sudo cp $files_path/ixgbe.ko $FILESYSTEM_ROOT/lib/modules/${LINUX_KERNEL_VERSION}-amd64/kernel/drivers/net/ethernet/intel/ixgbe/ixgbe.ko
-fi
-
```
Comment (jleveque) [inline build_debian.sh:191]: I take it the latest version of the driver is now included in the 4.19 kernel. Is this correct?

File: build_debian.sh [hunk 12]
```diff
@@ -282,14 +277,16 @@ sudo LANG=C DEBIAN_FRONTEND=noninteractive chroot $FILESYSTEM_ROOT apt-get -y in
     makedumpfile            \
     conntrack               \
     python-pip              \
-    python3-pip
+    python3-pip             \
+    cron                    \
+    haveged
```
Comment (jleveque) [inline build_debian.sh:282]: Why is `haveged` needed?

File: Makefile [hunk 13]
```diff
@@ -1,23 +1,36 @@
...
+NOSTRETCH ?= 0
```
Comment (qiluo-msft) [inline Makefile:4]: >NOSTRETCH [](start = 0, length = 9)

The NOSOMETHING macros are counter-intuitive. Seems more and more coming.

File: platform/barefoot/bfn-modules/debian/control [hunk 14]
```diff
@@ -7,6 +7,5 @@ Standards-Version: 3.9.3
 
 Package: bfn-modules
 Architecture: amd64
-Depends: linux-image-4.9.0-11-2-amd64
```
Comment (qiluo-msft) [inline platform/barefoot/bfn-modules/debian/control:10]: >linux-image-4.9.0-11-2-amd64 [](start = 9, length = 28)

It should depend on another version of linux-image?

File: platform/broadcom/sonic-platform-modules-accton/as4630-54pe/modules/x86-64-accton-as4630-54pe-cpld.c [hunk 15]
```diff
@@ -36,6 +36,7 @@
 #include <linux/delay.h>
 #include <linux/jiffies.h>
 #include <linux/kernel.h>
+#include <linux/fs.h>
```
Comment (qiluo-msft) [inline platform/broadcom/sonic-platform-modules-accton/as4630-54pe/modules/x86-64-accton-as4630-54pe-cpld.c:39]: Is it related to buster? If not, separate it into another PR?

File: platform/broadcom/sonic-platform-modules-accton/as5712-54x/modules/leds-accton_as5712_54x.c [hunk 16]
```diff
@@ -34,7 +34,6 @@ extern int as5712_54x_cpld_read (unsigned short cpld_addr, u8 reg);
 extern int as5712_54x_cpld_write(unsigned short cpld_addr, u8 reg, u8 value);
 
 extern void led_classdev_unregister(struct led_classdev *led_cdev);
-extern int led_classdev_register(struct device *parent, struct led_classdev *led_cdev);
```
Comment (qiluo-msft) [inline platform/broadcom/sonic-platform-modules-accton/as5712-54x/modules/leds-accton_as5712_54x.c:37]: Is it related to buster? If not, separate it into another PR?

File: platform/mellanox/rules.mk [hunk 17]
```diff
@@ -28,6 +28,3 @@ $(SYNCD)_RDEPENDS += $(MLNX_SAI)
 
 # Inject mlnx sdk libs to platform monitor
 $(DOCKER_PLATFORM_MONITOR)_DEPENDS += $(APPLIBS) $(SX_COMPLIB) $(SXD_LIBS) $(SX_GEN_UTILS) $(PYTHON_SDK_API) $(APPLIBS_DEV) $(SX_COMPLIB_DEV) $(SXD_LIBS_DEV) $(SX_GEN_UTILS_DEV)
-
-# Inject mlnx mlx libs to platform monitor
-$(DOCKER_PLATFORM_MONITOR)_DEPENDS += $(MFT)
```
Comment (qiluo-msft) [inline platform/mellanox/rules.mk:33]: Why remove?

File: rules/python-click.mk [hunk 18]
```diff
@@ -7,11 +7,10 @@
 # TODO: If we upgrade to a distro which provides a version >= 6.7 we will no
 # longer need to build this.
 
-PYTHON_CLICK_VERSION = 6.7-4
+PYTHON_CLICK_VERSION = 7.0-1
```
Comment (jleveque) [inline rules/python-click.mk:10]: Per my comment above, if the distro provides python-cli .deb package >= 6.7, it will contain the bug fixes we need, so we no longer need to build the package from source. Buster provides version 7.0-1.

File: rules/python-click.mk [hunk 19]
```diff
@@ -7,11 +7,10 @@
 # TODO: If we upgrade to a distro which provides a version >= 6.7 we will no
 # longer need to build this.
```
Comment (qiluo-msft) [inline rules/python-click.mk:8]: Seems no need to build

File: scripts/build_kvm_image.sh [hunk 20]
```diff
@@ -51,6 +51,7 @@ create_disk
 prepare_installer_disk
 
 echo "Prepare memory for KVM build: $vs_build_prepare_mem"
+sudo mount proc /proc -t proc || true
```
Comment (qiluo-msft) [inline scripts/build_kvm_image.sh:54]: Is it related to buster? If not, separate it into another PR?

File: rules/python-click.mk (hunk 19 above)
Comment (jleveque) [reply rules/python-click.mk:8]: @qiluo-msft: I addressed this in a separate comment. You can delete or resolve this comment, as it is a duplicate.

File: sonic-slave-buster/sonic-jenkins-id_rsa.pub [hunk 21]
```diff
@@ -0,0 +1 @@
+ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC769BQUJVeSIOyPsN4/Vo8xTqXQ6RI7ysVyCw/ABP3FIxf+fxmtm8t/Nbp9hq0uLHOjCw8UQbJ+XltsThFWJfH6RJY5NbfvwG7nUDjfjjp+SGEIHaVgIlpiuqiPbZ6QMjZ8Q0Sgi5p5ts1xe/4TFThwOJBHmhwydD5nk3BH7P3DDwlOCov5gjM40uMZJkketlO83zGG+25zu7O0hfDVt1vyK9bNWAhhPmGc79zdetfeFCxjimsff2m31B1KuVXiT5PDB1w+BSrUK6nNzJubnYCRgjg4prVTjA50EhlT2P7EoJAbW3TnTq8vUDkcstsGe/HZpfIB1VHBX97u4fAfGJZ root@acs-jenkins
```
Comment (qiluo-msft) [inline sonic-slave-buster/sonic-jenkins-id_rsa.pub:1]: >ssh-rsa AAAAB3 [](start = 0, length = 14)

Use a symlink file?

**Comment by qiluo-msft** [comment]
> https://bugs.debian.org/cgi-bin/bugreport.cgi?att=0;bug=468114;msg=32

The file content is expected to be same as this link. So will you update the link at the same time?

---
Refers to: src/initramfs-tools/loopback-file-system-support.patch:1 in 215a2e6. [](commit_id = 215a2e6bf4c7c28ded8bd99d2d498d2c1150b5e2, deletion_comment = False)

File: src/initramfs-tools/Makefile [hunk 22]
```diff
@@ -5,12 +5,12 @@ SHELL = /bin/bash
 MAIN_TARGET = initramfs-tools_$(INITRAMFS_TOOLS_VERSION)_all.deb
 DERIVED_TARGETS = initramfs-tools-core_$(INITRAMFS_TOOLS_VERSION)_all.deb
 
-INITRAMFS_TOOLS_REVISION = 18fc98e1b63b012f9bcf06ae3f5477872a5880c0
+INITRAMFS_TOOLS_REVISION = 40e544e
```
Comment (qiluo-msft) [inline src/initramfs-tools/Makefile:8]: >40e544e [](start = 27, length = 7)

Use a long hash? It may be invalid if the upstream grows.

File: src/sonic-config-engine/tests/sample_output/mvrf_interfaces [hunk 23]
```diff
@@ -29,13 +29,10 @@ iface eth0 inet static
     up ip -4 route add default via 10.0.0.1 dev eth0 table 5000 metric 201
     up ip -4 route add 10.0.0.0/24 dev eth0 table 5000
     up ip -4 rule add from 10.0.0.100/32 table 5000
-    up cgcreate -g l3mdev:mgmt
-    up cgset -r l3mdev.master-device=mgmt mgmt
```
Comment (qiluo-msft) [inline src/sonic-config-engine/tests/sample_output/mvrf_interfaces:33]: Is it related to buster? If not, separate it into another PR?

File: src/tacacs/nss/0006-fix-compiling-warning-about-token-dereference.patch [hunk 24]
```diff
@@ -0,0 +1,34 @@
...
+diff --git a/nss_tacplus.c b/nss_tacplus.c
+index 3ff3c35..7feeda6 100644
+--- a/nss_tacplus.c
++++ b/nss_tacplus.c
+@@ -87,7 +87,7 @@ static int parse_tac_server(char *srv_buf)
+ 
+     token = strsep(&srv_buf, delim);
+     while(token) {
+-        if('\0' != token) {
++        if('\0' != token[0]) {
```
Comment (qiluo-msft) [inline src/tacacs/nss/0006-fix-compiling-warning-about-token-dereference.patch:19]: >if('\0' != token[0]) { [](start = 9, length = 22)

Seems a good patch for non-buster.

File: Makefile (hunk 10 above)
Comment (lguohan) [reply Makefile:3]: not yet. still keep jessie here in case people wants to develop jessie based docker.

File: Makefile (hunk 13 above)
Comment (lguohan) [reply Makefile:4]: this is not related to buster upgrade. will triage this in the normal dev process, not bundle to buster upgrade.

File: build_debian.sh (hunk 11 above)
Comment (lguohan) [reply build_debian.sh:191]: yes, it is newer than 5.2.4

File: build_debian.sh (hunk 12 above)
Comment (lguohan) [reply build_debian.sh:282]: check the commit message here. https://github.com/Azure/sonic-buildimage/pull/4405/commits/2bdb348b92e5d0add128c1960d3d8f111e5466f8

File: build_debian.sh (hunk 9 above)
Comment (lguohan) [reply build_debian.sh:415]: yes, we have upgrade ansible to 2.8.9 which uses docker instead of docker-py

File: sonic-slave-buster/sonic-jenkins-id_rsa.pub (hunk 21 above)
Comment (lguohan) [reply sonic-slave-buster/sonic-jenkins-id_rsa.pub:1]: symbol link does not work docker build

**Comment by lguohan** [comment]
retest broadcom please

File: platform/barefoot/bfn-modules/debian/control (hunk 14 above)
Comment (lguohan) [reply platform/barefoot/bfn-modules/debian/control:10]: I just remove this line so that we do not need to change when we upgrade the kernel.

File: platform/broadcom/sonic-platform-modules-accton/as4630-54pe/modules/x86-64-accton-as4630-54pe-cpld.c (hunk 15 above)
Comment (lguohan) [reply platform/broadcom/sonic-platform-modules-accton/as4630-54pe/modules/x86-64-accton-as4630-54pe-cpld.c:39]: yes, it is needed.

File: platform/broadcom/sonic-platform-modules-accton/as5712-54x/modules/leds-accton_as5712_54x.c (hunk 16 above)
Comment (lguohan) [reply platform/broadcom/sonic-platform-modules-accton/as5712-54x/modules/leds-accton_as5712_54x.c:37]: yes

File: platform/mellanox/rules.mk (hunk 17 above)
Comment (lguohan) [reply platform/mellanox/rules.mk:33]: no longer needed by mellanox.

File: rules/python-click.mk (hunk 18 above)
Comment (lguohan) [reply rules/python-click.mk:10]: to make this pr simple, this will be address after buster merge.

File: platform/barefoot/bfn-modules/debian/control (hunk 14 above)
Comment (qiluo-msft) [reply platform/barefoot/bfn-modules/debian/control:10]: I know it is convenient. But is it the right thing to do? 

---
In reply to: [406544761](https://github.com/Azure/sonic-buildimage/pull/4405#discussion_r406544761) [](ancestors = 406544761)

File: src/tacacs/nss/0006-fix-compiling-warning-about-token-dereference.patch (hunk 24 above)
Comment (lguohan) [reply src/tacacs/nss/0006-fix-compiling-warning-about-token-dereference.patch:19]: have a separate pr.

File: scripts/build_kvm_image.sh (hunk 20 above)
Comment (lguohan) [reply scripts/build_kvm_image.sh:54]: anyway, I plan not to squash merge on this pr. I plan to do a merge on this. is already a separate commit. 3df2831a9ff6572f9843c8aa793eebe7c356be1d

File: files/build_templates/sonic_debian_extension.j2 (hunk 4 above)
Comment (lguohan) [reply files/build_templates/sonic_debian_extension.j2]: the problem is /proc is not not mount. Now, I mount /proc in sonic-slave docker and now umount outside fsroot is ok now.

File: platform/barefoot/bfn-modules/debian/control (hunk 14 above)
Comment (lguohan) [reply platform/barefoot/bfn-modules/debian/control:10]: all the modules are built in the context of sonic buildimage, therefore we do not have such issue to install the packages for other kernel.

File: src/initramfs-tools/Makefile (hunk 22 above)
Comment (lguohan) [reply src/initramfs-tools/Makefile]: done

File: src/tacacs/nss/0006-fix-compiling-warning-about-token-dereference.patch (hunk 24 above)
Comment (lguohan) [reply src/tacacs/nss/0006-fix-compiling-warning-about-token-dereference.patch]: done

File: src/sonic-config-engine/tests/sample_output/mvrf_interfaces (hunk 23 above)
Comment (lguohan) [reply src/sonic-config-engine/tests/sample_output/mvrf_interfaces:33]: it is related.

**Comment by lguohan** [comment]
retest baseimage please

**Comment by lguohan** [comment]
retest baseimage please

File: files/build_templates/sonic_debian_extension.j2 [hunk 25]
```diff
@@ -344,7 +344,14 @@ sudo chmod a+x $FILESYSTEM_ROOT/usr/sbin/policy-rc.d
 {% if installer_debs.strip() -%}
 {% for deb in installer_debs.strip().split(' ') -%}
 {% if sonic_asic_platform == "mellanox" %}
-sudo dpkg --extract {{deb}} $FILESYSTEM_ROOT
```
Comment (qiluo-msft) [inline files/build_templates/sonic_debian_extension.j2:347]: What is the purpose? related to buster?

File: Makefile.work [hunk 26]
```diff
@@ -76,11 +76,16 @@ ifeq ($(PLATFORM_ARCH),)
 	override PLATFORM_ARCH = $(CONFIGURED_ARCH)
 endif
 
+ifeq ($(BLDENV), buster)
+SLAVE_DIR = sonic-slave-buster
+else
 ifeq ($(BLDENV), stretch)
```
Comment (jleveque) [inline Makefile.work:82]: Suggest to combine these into one line, e.g, `else ifeq ($(BLDENV), stretch)` and remove the extra `endif` on line 87. I believe this syntax should be supported by our version of make.

File: files/build_templates/sonic_debian_extension.j2 (hunk 25 above)
Comment (lguohan) [reply files/build_templates/sonic_debian_extension.j2:347]: check this pr. https://github.com/Azure/sonic-buildimage/pull/4405/commits/9b9e5c72640ff4c322e9a784ae63677e09584239

it is a workaround. somehow, the dpkg extract corrupt the root file system. I think mellanox own an official fix for this. they are working on this.

File: Makefile.work (hunk 26 above)
Comment (lguohan) [reply Makefile.work:82]: thansk. resolved. will push.

File: platform/barefoot/bfn-modules/debian/control (hunk 14 above)
Comment (lguohan) [reply platform/barefoot/bfn-modules/debian/control:10]: i restored those depends.

**Comment by lguohan** [comment]
retest vsimage please

**Comment by lguohan** [comment]
> Adding the BLDENV to SONIC_COMMON_FLAGS_LIST would help to resolve the package generation based on the base image type?.
> 
> SONIC_COMMON_FLAGS_LIST += $(BLDENV)

Not quite, because other packages such as DOCKER, it does not make a difference whether it is built from buster or stretch.

**Comment by stephenxs** [comment]
@lguohan @jleveque I think it also depends on the sonic-utilities PR [[201911] Explicitly specify command names with underscores (continued)](https://github.com/Azure/sonic-utilities/pull/868). 
Can you advance the submodule head?
Thanks.

File: src/sonic-config-engine/tests/sample_output/mvrf_interfaces (hunk 23 above)
Comment (irene-pan1202) [reply src/sonic-config-engine/tests/sample_output/mvrf_interfaces:33]: @lguohan Are these corrections because the 4.19 kernel has some issues? mgmt vrf must use cgexec command, which means mgmt vrf is not supported in 4.19 kernel?
